### PR TITLE
fix(core): Fix corner cases in initialized range tracking

### PR DESCRIPTION
**Pull request by andyleiserson**
Adds a [proptest](https://docs.rs/proptest/latest/proptest/) for the uninitialized range tracker, and fixes a few corner cases.

This does add a new dev-dependency on proptest; I think these are good tests and this probably isn't the only place where we could benefit, but every dependency costs something.

**Testing**
The proptest; also includes directed tests for the fixes.

**Squash or Rebase?** Squash

**Checklist**

- [x] I self-reviewed and fully understand this PR.
- [x] WebGPU implementations built with `wgpu` may be affected behaviorally.
- [ ] Validation and feature gates are in place to confine behavioral changes.
- [x] Tests demonstrate the validation and altered logic works. 
- [x] `CHANGELOG.md` entries for the user-facing effects of this change are present. 
- [x] The PR is minimal, and doesn't make sense to land as multiple PRs.
- [ ] Commits are logically scoped and individually reviewable.
- [x] The PR description has enough context to understand the motivation and solution implemented.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -178,6 +178,8 @@ portable-atomic = "1.10"
 portable-atomic-util = "0.2.4"
 pp-rs = "0.2.1"
 profiling = { version = "1.0.1", default-features = false }
+# Limit to versions that use rand 0.8
+proptest = ">=1, <1.7"
```
Comment (andyleiserson) [inline Cargo.toml:182]: It looks like our use of `rand` 0.8 is via `noise`, for the `water` example.

Comment (ErichDonGubler) [reply Cargo.toml:182]: Looks like `noise` uses 0.9 in upstream (see https://github.com/Razaekel/noise-rs/blob/77de33887620c2a7e7263b1d30ef46afbfe705b6/Cargo.toml#L19), but it hasn't been released yet.

**Review by ErichDonGubler — APPROVED** [review]
Nice!
